lib/task.js

```diff
@@ -180,18 +201,27 @@ function factory(
         }
 
         var rendered;
+        var deferredRender = false;
 
         // Support OR syntax: {{ options.value | options.backupValue | default }}
         // ^^ Will return "default" if options.value and options.backupValue do not exist
         _.some(str.split(/[\s\n]?\|+[\s\n]?/), function(item) {
-            // If item is a string that should not be rendered yet, defer rendering
-            // until the next render stage
-            rendered = item.indexOf('.') > -1 ? catalogSearch.getPath(context, item) : item;
+            // If item is a string that should not be rendered yet (i.e. it requires
+            // data generated dynamically at workflow runtime), defer rendering
+            // until later
+            if (self._isDeferredRender(item.split('.')[0], self.state)) {
```

Comment:
I'll remove it. I got part of this code from a 6 month old version of the code that existed before we initially removed deferred rendering, and left some pieces in that should be removed.